src/Aspire.Hosting/Dashboard/DcpDataSource.cs

```diff
@@ -36,22 +37,46 @@ internal sealed class DcpDataSource
     public DcpDataSource(
         IKubernetesService kubernetesService,
         DistributedApplicationModel applicationModel,
+        DistributedApplicationExecutionContext executionContext,
         IConfiguration configuration,
         ILoggerFactory loggerFactory,
         Func<ResourceSnapshot, ResourceSnapshotChangeType, ValueTask> onResourceChanged,
         CancellationToken cancellationToken)
     {
         _kubernetesService = kubernetesService;
-        _applicationModel = applicationModel;
+        _applicationModel = applicationModel.Resources.ToDictionary(d => d.Name);
+        _executionContext = executionContext;
         _onResourceChanged = onResourceChanged;
 
         _logger = loggerFactory.CreateLogger<DcpDataSource>();
 
+        var initialTcs = new TaskCompletionSource();
+
+        // Show all resources initially and allow updates from DCP (for the relevant resources)
+        Task.Run(async () =>
+        {
+            try
+            {
+                foreach (var resource in applicationModel.Resources)
+                {
+                    await ProcessInitialResource(resource).ConfigureAwait(false);
+                }
+            }
+            finally
+            {
+                // Start watching resources after we have loaded the initial state
+                initialTcs.TrySetResult();
```

Comment:
Why use a TCS instead of getting the task returned from Task.Run and awaiting that?